setup.sh

```
@@ -1,39 +1,197 @@
+# Try to install to system locations
+INSTALLED=false
+INSTALL_LOCATIONS=(
+    "/usr/local/bin"
+    "$HOME/.local/bin"
+    "$HOME/bin"
+)
+
+for loc in "${INSTALL_LOCATIONS[@]}"; do
+    if [ -d "$loc" ] && [ -w "$loc" ]; then
+        ln -sf "$(pwd)/src/madspark/bin/mad_spark" "$loc/mad_spark" 2>/dev/null
+        if [ $? -eq 0 ]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)